src/backend/distributed/metadata/dependency.c

```diff
@@ -1515,6 +1631,36 @@ ExpandCitusSupportedTypes(ObjectAddressCollector *collector, ObjectAddress targe
 }
 
 
+/*
+ * ExpandForPgVanilla only expands only comosite types because other types
+ * will find their dependencies in pg_depend. The method should only be called by
+ * is_citus_depended_object udf.
+ */
+static List *
+ExpandForPgVanilla(ObjectAddressCollector *collector,
+				   ObjectAddress target)
+{
+	List *result = NIL;
+
```

Comment:
should we add the assert that GUC should be true here as well? To prevent future callers to use